CONTRIBUTING.md

```diff
@@ -173,3 +207,124 @@ locations.
     └─ ...
  └─ ...
 ```
+
+## PHP internals
+
+For information on PHP internal C functions see
+[PHP at the Core: A Hacker's Guide](https://php.net/internals), though this is
+considered incomplete. Various external resources can be found on the web. A
+standard printed reference is the book "Extending and Embedding PHP" by Sara
+Golemon.
+
+## Submitting patches
```

Comment:
I think we should just remove this section entirely and encourage people to send PRs only. Patches by mail are an annoyance and cannot be reasonably reviewed.